### PR TITLE
E93 upload

### PR DESCRIPTION
**Pull request by dudiq**

### CODE REVIEW

File: src/components/forms/DocumentUploadForm/DocumentUploadProgress.vue [hunk 1]
```diff
@@ -13,6 +13,8 @@
 <script>
 import VProgressCircle from '@endpass/ui/kit/VProgressCircle';
 
+const MIN_PROGRESS_VALUE = 2;
```
Comment (kazurus) [inline src/components/forms/DocumentUploadForm/DocumentUploadProgress.vue]: maybe, move that logic to the store?

File: src/components/forms/DocumentUploadForm/DocumentUploadArea/DocumentUploadFileStatus.vue [hunk 2]
```diff
@@ -13,16 +16,10 @@
         @click.native.prevent="$emit('remove', $event)"
       />
     </div>
-    <v-description
```
[the remark this answers is not in the document]
Comment (dudiq) [reply src/components/forms/DocumentUploadForm/DocumentUploadArea/DocumentUploadFileStatus.vue]: error can be without file, and we need show only error

File: src/components/forms/DocumentUploadForm/DocumentUploadProgress.vue (hunk 1 above)
Comment (dudiq) [reply src/components/forms/DocumentUploadForm/DocumentUploadProgress.vue]: this logic for view only, store must always return correct value

File: src/components/forms/DocumentUploadForm/DocumentUploadArea/DocumentUploadArea.vue [hunk 3]
```diff
@@ -10,14 +10,21 @@
...
       </div>
-      <document-upload-field
-        v-else-if="file"
-        :message="fileReady"
-        :error="error"
-        :file-name="file.name"
-        class="document-upload-field-file"
-        @remove="onRemoveFile"
-      />
+      <div v-else-if="file || error">
```
[the remark this answers is not in the document]
Comment (dudiq) [reply src/components/forms/DocumentUploadForm/DocumentUploadArea/DocumentUploadArea.vue]: there are three states:
- error
- file
- file and error

that's why i was separated to wrapper this block

[the remark this answers is not in the document]
Comment (kazurus) [reply src/components/forms/DocumentUploadForm/DocumentUploadArea/DocumentUploadArea.vue]: Will two blocks with `v-else-if` be the solution?

File: src/components/forms/DocumentUploadForm/UploadError.vue [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+export default {
+  name: 'UploadError',
+
+  props: {
+    file: {
+      type: File,
+      default: null,
+    },
+    error: {
```
Comment (kazurus) [inline src/components/forms/DocumentUploadForm/UploadError.vue]: isn't used

File: src/components/screens/DocumentCreate/DocumentUpload/DocumentUpload.vue [hunk 5]
```diff
@@ -1,171 +1,201 @@
...
-        return this.$t('components.uploadDocument.recognition');
-      }
-      return '';
-    },
     isLoading() {
-      const { uploadController } = this.$options;
-      return uploadController.isUploading || uploadController.isProcessing;
+      const { uploadController } = this;
+      return (
+        uploadController.isUploading ||
```
Comment (kazurus) [inline src/components/screens/DocumentCreate/DocumentUpload/DocumentUpload.vue]: maybe in store? e.g. isProcessing?

File: tests/unit/specs/components/screens/DocumentCreate/DocumentUpload/DocumentUpload.spec.js [hunk 6]
```diff
@@ -32,10 +33,37 @@ describe('DocumentUpload', () => {
...
+    wrapper.find('document-upload-buttons-stub').vm.$emit('upload');
+
+    await global.flushPromises();
+
+    wrapper.find('document-upload-buttons-stub').vm.$emit('done');
+
+    await global.flushPromises();
+
+    expect(wrapper.find('document-upload-form-stub').attributes().error).toBe(
+      'Something went wrong while document recognition. Please try again.',
```
Comment (kazurus) [inline tests/unit/specs/components/screens/DocumentCreate/DocumentUpload/DocumentUpload.spec.js]: i18n here and in other places

File: src/components/screens/DocumentCreate/DocumentUpload/DocumentUploadButtons.vue [hunk 7]
```diff
@@ -0,0 +1,94 @@
+<template>
+  <form-controls>
+    <v-button
```
Comment (kazurus) [inline src/components/screens/DocumentCreate/DocumentUpload/DocumentUploadButtons.vue]: dynamic?

**Pull request merged**